File: frontend/web_ccadmin_ecommerce/src/app/enterprise/sale/model/entity/TrxPaymentDocumentEntity.ts
import { AuditTableEntity } from '../../../shared/model/entity/AuditTableEntity';

export class TrxPaymentDocumentEntity extends AuditTableEntity {
  public TrxPaymentDocumentId: number | null = null;
  public TrxPaymentId: number | null = null;
  public DocumentType: string = '';
  public ContentEncoding: string = '';
  public Content: string = '';
  public FileName: string = '';
  public ContentType: string = '';
  public SizeBytes: number | null = null;
  public Sha256Hash: string = '';
  public SourceType: string = '';
  public PurgeAfterDate: Date | null = null;
}

File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/payment/model/entity/TrxPaymentDocumentEntity.java
package com.ccadmin.app.payment.model.entity;

import com.ccadmin.app.shared.model.entity.AuditTableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Lob;
import jakarta.persistence.Table;

import java.io.Serializable;
import java.util.Date;

@Entity
@Table(name = "trx_payments_document")
public class TrxPaymentDocumentEntity extends AuditTableEntity implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long TrxPaymentDocumentId;
    public Long TrxPaymentId;
    public String DocumentType;
    public String ContentEncoding;
    @Lob
    @Column(columnDefinition = "longtext")
    public String Content;
    public String FileName;
    public String ContentType;
    public Long SizeBytes;
    public String Sha256Hash;
    public String SourceType;
    public Date PurgeAfterDate;
}
